docs/RFCS/20210415_db_console_cluster_ui.md

```diff
@@ -0,0 +1,74 @@
+
+
+## Updating DB Console
+
+After the code is moved, dependencies have been resolved, and the build is successful the native code that references the `@cockroachlabs/cluster-ui` dependency should be updated to point to local code instead. There are currently around 35 references to the external dependency with most of them being utility components. There will most likely be some component duplication when moving the whole of Cluster UI back into the repo from whence it came. There will be an opportunity to remove identical duplicate components, but I think an evaluation of similar components should be postponed until both applications have been fully updated. 
+
+Once the move and cleanup has happened on master, an update will need to be made to `release-20.2` as this version also depends on the an external `cluster-ui` as well. The version supporting 20.2 is on a separate branch in the UI repo (`cluster-ui-20.2`) and so while the code move on master could be referenced I'm not confident it can be duplicated. Depending on the state of the release, the work on CRDB master may need to be backported to `release-21.1` as well.
+
+## Updating CC Console
+
+When we have confirmed that both DB Console can be built successfully and the Cluster UI package can be build independently, publishing for Cluster UI should be configured. Currently in the UI repo, the `cluster-ui` package is published by a Github Action when pull requests are merged to master. Eventually it would be highly beneficial to have publishing be a part of merging code, we anticipate that immediately after move publishing will be a manual process identical to the publishing of `@cockroachlabs/crdb-protobuf-client`. New versions of `@cockroachlabs/cluster-ui` should be published to match the version of DB Console they represent (21.2.x, 21.1.x, 20.2.x). Documentation for the Cluster UI package should be updated to reflect any significant architectural changes (probably mostly related to protobufs) and describing the new versioning. When the new versions are available on npm, CockroachCloud console should be updated to depend on these new versions. I don't anticipate any other code changes necessary as this proposal does not include any change in functionality of the components.
+
+## Deprecate the package in UI repo
+
+As a final step, the code in `cockroachdb/ui` should be deprecated, the `package.json` set to private, and the Github Action for publishing removed. We should keep the code around for reference or back up until we are confident it is no longer needed.
```

Comment:
Here you mean just the cluster-ui package or all packages on the ui repo?

Reply:
Oh, just the `cluster-ui` package. I'll change that to be more explicit. We will keep all the other packages in that repo.